python/lsst/pipe/tasks/interpImage.py

```diff
@@ -168,7 +174,78 @@ def run(self, image, planeName=None, fwhmPixels=None, defects=None):
         if self.config.useFallbackValueAtEdge:
             fallbackValue = self._setFallbackValue(maskedImage)
 
-        measAlg.interpolateOverDefects(maskedImage, psf, defectList, fallbackValue,
-                                       self.config.useFallbackValueAtEdge)
+        self.interpolateImage(maskedImage, psf, defectList, fallbackValue)
 
         self.log.info("Interpolated over %d %s pixels." % (len(defectList), planeName))
+
+    @contextmanager
+    def transposeContext(self, maskedImage, defects):
+        """Context manager to potentially transpose an image
+
+        This applies the ``transpose`` configuration setting.
+
+        Transposing the image allows us to interpolate along columns instead
+        of rows, which is useful when the saturation trails are typically
+        oriented along rows on the warped/coadded images, instead of along
+        columns as they typically are in raw CCD images.
+
+        Parameters
+        ----------
+        maskedImage : `lsst.afw.image.MaskedImage`
+            Image on which to perform interpolation.
+        defects : iterable of `lsst.afw.image.Defect`
+            List of defects to interpolate over.
+
+        Yields
+        ------
+        useImage : `lsst.afw.image.MaskedImage`
+            Image to use for interpolation; it may have been transposed.
+        useDefects : iterable of `lsst.afw.image.Defect`
+            List of defects to use for interpolation; they may have been
+            transposed.
+        """
+        def transposeBox(box):
+            """Transpose a bounding box"""
+            return afwGeom.Box2I(afwGeom.Point2I(box.getMinY(), box.getMinX()),
+                                 afwGeom.Point2I(box.getMaxY(), box.getMaxX()))
+
+        def transposeImage(image):
+            """Transpose an image"""
+            result = type(image)(transposeBox(image.getBBox()))
+            result.array[:] = image.array.T
+            return result
+
+        useImage = maskedImage
+        useDefects = defects
+        if self.config.transpose:
+            useImage = afwImage.makeMaskedImage(transposeImage(maskedImage.image),
+                                                transposeImage(maskedImage.mask),
+                                                transposeImage(maskedImage.variance))
+            useDefects = ipIsr.transposeDefectList(defects)
+        try:
```

Comment:
Unclear to me why you put this in a `try/finally` block here.  If the code in the `with` block that uses this context throws an exception, yield rethrows it (regardless if the try block is here or not). If that happens, what's the point of putting the pre-interpolation pixels back in?  This isn't restoring some previous state.